diff --git a/src/openvpn/forward.c b/src/openvpn/forward.c
index 5004e354..6907afb3 100644
--- a/src/openvpn/forward.c
+++ b/src/openvpn/forward.c
@@ -936,7 +936,12 @@ read_incoming_link(struct context *c, struct link_socket *sock)
     c->c2.buf = c->c2.buffers->read_link_buf;
     ASSERT(buf_init(&c->c2.buf, c->c2.frame.buf.headroom));
 
-    status = link_socket_read(sock, &c->c2.buf, &c->c2.from);
+    status = link_socket_read(sock,
+                              &c->c2.buf,
+                              &c->c2.from,
+                              c->options.ce.xormethod,
+                              c->options.ce.xormask,
+                              c->options.ce.xormasklen);
 
     if (socket_connection_reset(sock, status))
     {
@@ -1806,7 +1811,12 @@ process_outgoing_link(struct context *c, struct link_socket *sock)
                 socks_preprocess_outgoing_link(c, sock, &to_addr, &size_delta);
 
                 /* Send packet */
-                size = (int)link_socket_write(sock, &c->c2.to_link, to_addr);
+                size = (int)link_socket_write(sock,
+                                              &c->c2.to_link,
+                                              to_addr,
+                                              c->options.ce.xormethod,
+                                              c->options.ce.xormask,
+                                              c->options.ce.xormasklen);
 
                 /* Undo effect of prepend */
                 link_socket_write_post_size_adjust(&size, size_delta, &c->c2.to_link);
